sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -642,8 +642,15 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
       if (stats.get.rowCount.isDefined) {
         statsProperties += STATISTICS_NUM_ROWS -> stats.get.rowCount.get.toString()
       }
+
+      // For datasource tables the data schema is stored in the table properties.
+      val schema = rawTable.properties.get(DATASOURCE_PROVIDER) match {
+        case Some(provider) => getSchemaFromTableProperties(rawTable)
+        case _ => rawTable.schema
```

Comment:
@viirya right. I agree. I was saying that we do have a raw table from a prior call. So here we just pass that to restoreTableMetadata like you suggested.